### PR TITLE
prelim

### PR DESCRIPTION
**Pull request by sahma19**
Update Cvr API, to use lists of cvr numbers, so querying for multiple cvr numbers, in one request is possible

### CODE REVIEW

File: domains/transfer/Transfer.API/API.IntegrationTests/Cvr/Api/v2024_01_03/Controllers/CvrMultipleResponseTests.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    [Fact]
+    public async Task GetCvrCompany_WhenWrongPlusRight_ShouldReturnRightOnly()
+    {
+        var wrongPlusRightCvr = new List<string> { "123", "28980671", "39315041" };
+        server.ResetMappings();
+        server
+            .Given(Request.Create().WithPath("/cvr-permanent/virksomhed/_search").UsingPost())
+            .RespondWith(Response.Create()
+                .WithStatusCode(200)
+                .WithBodyFromFile("Cvr/Api/v2024_01_03/Controllers/CvrControllerTests.cvr_response2.json")
```
Comment (ckr123) [inline domains/transfer/Transfer.API/API.IntegrationTests/Cvr/Api/v2024_01_03/Controllers/CvrMultipleResponseTests.cs:40]: What is response2?

File: domains/transfer/Transfer.API/API/Cvr/Api/v2024_01_03/Controllers/CvrController.cs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            return Ok(new CvrCompanyListResponse(new List<CvrCompanyDto>()));
+
+        var searchResult = await client.CvrNumberSearch(parsedCvrNumbers);
+        if (searchResult?.hits?.hits == null)
+            return Ok(new CvrCompanyListResponse(new List<CvrCompanyDto>()));
+
+        var companies = searchResult.hits.hits
+            .Select(hit => ToDto(hit._source?.Vrvirksomhed))
+            .Where(dto => dto != null)
+            .ToList();
```
Comment (tnickelsen) [inline domains/transfer/Transfer.API/API/Cvr/Api/v2024_01_03/Controllers/CvrController.cs:41]: Put this in a ToDto function

File: domains/transfer/Transfer.API/API/Program.cs [hunk 3]
```diff
@@ -46,7 +46,6 @@
 
 builder.Services.AddValidatorsFromAssembly(typeof(Program).Assembly);
 
-builder.Services.AddEndpointsApiExplorer();
```
Comment (tnickelsen) [inline domains/transfer/Transfer.API/API/Program.cs:49]: Why this deletion?

File: domains/transfer/ClaimAutomation/Worker.IntegrationTests/Worker.IntegrationTests.csproj [hunk 4]
```diff
@@ -19,11 +19,11 @@
...
     <PackageReference Include="FluentAssertions" Version="6.12.0" />
-    <PackageReference Include="Microsoft.AspNetCore.Mvc.Testing" Version="7.0.5" />
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.6.0" />
-    <PackageReference Include="Testcontainers.PostgreSql" Version="3.2.0" />
+    <PackageReference Include="Microsoft.AspNetCore.Mvc.Testing" Version="8.0.0" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.8.0" />
+    <PackageReference Include="Testcontainers.PostgreSql" Version="3.6.0" />
     <PackageReference Include="Verify.Xunit" Version="20.4.0" />
-    <PackageReference Include="xunit" Version="2.4.2" />
+    <PackageReference Include="xunit" Version="2.6.4" />
```
[the remark this answers is not in the document]
Comment (sahma19) [reply domains/transfer/ClaimAutomation/Worker.IntegrationTests/Worker.IntegrationTests.csproj:26]: Tænker en opdatering som ikke breaker noget, er harmløs. Dette gør også livet nemmere når vi skal lave dette issue:

https://app.zenhub.com/workspaces/team-atlas-633199659e255a37cd1d144f/issues/gh/energinet-datahub/energy-origin-issues/1868

File: domains/transfer/Transfer.API/API.IntegrationTests/Shared/Extensions/MockHttpMessageHandler.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace API.IntegrationTests.Shared.Extensions;
+
+public class MockHttpMessageHandler(string response, HttpStatusCode statusCode) : HttpMessageHandler
```
[the remark this answers is not in the document]
Comment (sahma19) [reply domains/transfer/Transfer.API/API.IntegrationTests/Shared/Extensions/MockHttpMessageHandler.cs]: Den er blevet fjernet igen

File: domains/transfer/Transfer.API/API.UnitTests/Transfer/Api/Converters/Base64ConverterTests.cs [hunk 6]
```diff
@@ -30,7 +30,7 @@ public void Base64Converter_ShouldBeAbleToConvertBackAfterConversion()
     }
 
     [Theory]
-    [InlineData(null)]
```
[the remark this answers is not in the document]
Comment (sahma19) [reply domains/transfer/Transfer.API/API.UnitTests/Transfer/Api/Converters/Base64ConverterTests.cs:33]: Opdatering af xUnit til nyeste version, kom med en ny linter, som foreslår at man ikke bruger null værdier.

File: domains/transfer/Transfer.API/API/Cvr/Api/v2024_01_03/Controllers/CvrController.cs [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+using System.Linq;
+using System.Threading.Tasks;
+using API.Cvr.Api.Clients.Cvr;
+using API.Cvr.Api.Models;
+using API.Cvr.Api.v2024_01_03.Dto.Responses;
+using Asp.Versioning;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace API.Cvr.Api.v2024_01_03.Controllers;
```
[the remark this answers is not in the document]
Comment (sahma19) [reply domains/transfer/Transfer.API/API/Cvr/Api/v2024_01_03/Controllers/CvrController.cs:11]: <3

File: domains/transfer/Transfer.API/API.IntegrationTests/Cvr/Api/v2024_01_03/Controllers/CvrMultipleResponseTests.cs [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+using FluentAssertions;
+using VerifyTests;
+using VerifyXunit;
+using WireMock.RequestBuilders;
+using WireMock.ResponseBuilders;
+using WireMock.Server;
+using Xunit;
+
+namespace API.IntegrationTests.Cvr.Api.v2024_01_03.Controllers;
+//This should be in the same class as CvrControllerTests, but WireMock does not Co-operate with that.
```
Comment (ckr123) [inline domains/transfer/Transfer.API/API.IntegrationTests/Cvr/Api/v2024_01_03/Controllers/CvrMultipleResponseTests.cs:17]: Why is that?

File: domains/transfer/Transfer.API/API/Cvr/Api/v2024_01_03/Controllers/CvrController.cs (hunk 2 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API/Cvr/Api/v2024_01_03/Controllers/CvrController.cs:41]: ToDto function, er kun beregnet til 1x dto, denne logik omhandler at putte flere dto'er i en liste. Ved ikke om toDto er det rigtige sted den skal være. en separat metode så?

File: domains/transfer/Transfer.API/API/Cvr/Api/v2024_01_03/Controllers/CvrController.cs [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+[Route("api/cvr")]
+public class CvrController(CvrClient client) : Controller
+{
+    /// <summary>
+    /// Get CVR registered company information for multiple CVR numbers
+    /// </summary>
+    /// <response code="200">Successful operation</response>
+    /// <response code="404">CVR company not found</response>
+    [ProducesResponseType(typeof(CvrCompanyListResponse), 200)]
+    [ProducesResponseType(typeof(void), 404)]
```
Comment (ckr123) [inline domains/transfer/Transfer.API/API/Cvr/Api/v2024_01_03/Controllers/CvrController.cs:25]: It returns 200 OK with an empty collection if cvr company not found

File: domains/transfer/Transfer.API/API.IntegrationTests/API.IntegrationTests.csproj [hunk 10]
```diff
@@ -50,5 +50,14 @@
 		<None Update="Cvr\Api\v2023_01_01\Controllers\CvrControllerTests.empty_cvr_response.json">
 		  <CopyToOutputDirectory>Always</CopyToOutputDirectory>
 		</None>
+        <None Update="Cvr\Api\v2024_01_03\Controllers\CvrControllerTests.cvr_response.json">
+            <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+        </None>
+        <None Update="Cvr\Api\v2024_01_03\Controllers\CvrControllerTests.empty_cvr_response.json">
+            <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+        </None>
+        <None Update="Cvr\Api\v2024_01_03\Controllers\CvrControllerTests.cvr_response2.json">
```
[the remark this answers is not in the document]
Comment (sahma19) [reply domains/transfer/Transfer.API/API.IntegrationTests/API.IntegrationTests.csproj:59]: Mine tests fejler, hvis jeg ikke har den med, så det forstår jeg ikke

File: domains/transfer/Transfer.API/API/Program.cs (hunk 3 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API/Program.cs]: Leftover change, jeg glemte at revert. Har reverted igen nu, i dette commit:

https://github.com/Energinet-DataHub/energy-origin/pull/558/commits/6887727c54f064949cef39556bca9ba41485aa3f

File: domains/transfer/Transfer.API/API/Cvr/Api/v2024_01_03/Controllers/CvrController.cs [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+    [HttpPost]
+    public async Task<ActionResult<CvrCompanyListResponse>> GetCvrCompanies([FromBody] List<string> cvrNumbers)
+    {
+        var parsedCvrNumbers = cvrNumbers.Select(CvrNumber.TryParse).OfType<CvrNumber>().ToList();
+
+        if (!parsedCvrNumbers.Any())
+            return Ok(new CvrCompanyListResponse(new List<CvrCompanyDto>()));
+
+        var searchResult = await client.CvrNumberSearch(parsedCvrNumbers);
+        if (searchResult?.hits?.hits == null)
```
Comment (ckr123) [inline domains/transfer/Transfer.API/API/Cvr/Api/v2024_01_03/Controllers/CvrController.cs:35]: What is returned if the first hits is null?

File: domains/transfer/Transfer.API/API/Program.cs [hunk 12]
```diff
@@ -62,7 +61,8 @@
     .AddApiExplorer();
 
 builder.Services.AddTransient<IConfigureOptions<SwaggerGenOptions>, ConfigureSwaggerOptions>();
-builder.Services.AddSwaggerGen();
+builder.Services.AddSwaggerGen(options => options.OperationFilter<SwaggerDefaultValues>());
```
Comment (ckr123) [inline domains/transfer/Transfer.API/API/Program.cs]: Why is this changed?

File: domains/transfer/Transfer.API/API/Shared/Swagger/ConfigureSwaggerOptions.cs [hunk 13]
```diff
@@ -1,34 +1,22 @@
...
-    }
-
     public void Configure(SwaggerGenOptions options)
     {
         options.OperationFilter<SwaggerDefaultValues>();
         options.SupportNonNullableReferenceTypes();
         var xmlFilePath = Path.Combine(AppContext.BaseDirectory, "documentation.xml");
         options.IncludeXmlComments(xmlFilePath);
 
-        if (environment.IsDevelopment())
```
Comment (ckr123) [inline domains/transfer/Transfer.API/API/Shared/Swagger/ConfigureSwaggerOptions.cs:31]: Why is this removed?

File: domains/transfer/Transfer.API/API.IntegrationTests/API.IntegrationTests.csproj (hunk 10 above)
[the remark this answers is not in the document]
Comment (tnickelsen) [reply domains/transfer/Transfer.API/API.IntegrationTests/API.IntegrationTests.csproj:59]: Prøv at slette dem fra .csproj kun, men ikke fra selve filsystemet

File: domains/transfer/ClaimAutomation/Worker.IntegrationTests/Worker.IntegrationTests.csproj (hunk 4 above)
[the remark this answers is not in the document]
Comment (tnickelsen) [reply domains/transfer/ClaimAutomation/Worker.IntegrationTests/Worker.IntegrationTests.csproj:26]: Det er så fint. Bare lige få xunit.runner.visualstudio med også

File: domains/transfer/Transfer.API/API/Transfer/Api/v2024_01_03/Dto/Responses/TransferAutomationStatus.cs [hunk 14]
```diff
@@ -0,0 +1,3 @@
+namespace API.Transfer.Api.v2024_01_03.Dto.Responses;
+
+public record TransferAutomationStatus(bool Healthy);
```
Comment (ckr123) [inline domains/transfer/Transfer.API/API/Transfer/Api/v2024_01_03/Dto/Responses/TransferAutomationStatus.cs:3]: This looks like some legaacy and can be removed

File: domains/transfer/Transfer.API/API.UnitTests/Transfer/Api/Converters/Base64ConverterTests.cs (hunk 6 above)
[the remark this answers is not in the document]
Comment (tnickelsen) [reply domains/transfer/Transfer.API/API.UnitTests/Transfer/Api/Converters/Base64ConverterTests.cs:33]: Sjovt. For det er jo en reel værdi til string. Hvad siger den til dig?

Comment (ckr123) [inline domains/transfer/Transfer.API/API.UnitTests/Transfer/Api/Converters/Base64ConverterTests.cs:33]: Why not null anymore?

[the remark this answers is not in the document]
Comment (sahma19) [reply domains/transfer/Transfer.API/API.UnitTests/Transfer/Api/Converters/Base64ConverterTests.cs:33]: image

File: domains/transfer/Transfer.API/API/Cvr/Api/v2024_01_03/Controllers/CvrController.cs (hunk 2 above)
Comment (tnickelsen) [reply domains/transfer/Transfer.API/API/Cvr/Api/v2024_01_03/Controllers/CvrController.cs:41]: Du kan vel bare have flere ToDto metoder, der har forskellige input parametre?

File: domains/transfer/Transfer.API/API/Cvr/Api/Clients/Cvr/CvrClient.cs [hunk 15]
```diff
@@ -15,21 +17,28 @@ public CvrClient(HttpClient client)
         this.client = client;
     }
 
-    public async Task<Root?> CvrNumberSearch(CvrNumber cvr)
+    public Task<Root?> CvrNumberSearch(CvrNumber cvr)
```
Comment (ckr123) [inline domains/transfer/Transfer.API/API/Cvr/Api/Clients/Cvr/CvrClient.cs:20]: I think this should be deleted and all the controllers that uses CvrNumberSearch should just use a list instead

File: domains/transfer/Transfer.API/API.IntegrationTests/Cvr/Api/v2024_01_03/Controllers/CvrMultipleResponseTests.cs (hunk 1 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API.IntegrationTests/Cvr/Api/v2024_01_03/Controllers/CvrMultipleResponseTests.cs:40]: a json document containing an elasticsearch response body, with 2 companies.

File: domains/transfer/Transfer.API/API.IntegrationTests/Cvr/Api/v2024_01_03/Controllers/CvrMultipleResponseTests.cs (hunk 8 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API.IntegrationTests/Cvr/Api/v2024_01_03/Controllers/CvrMultipleResponseTests.cs:17]: Tests, using wiremock, share resources, so the test always fail when running in parallel with the others. this is the fix suggested by @tnickelsen

File: domains/transfer/Transfer.API/API/Program.cs (hunk 12 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API/Program.cs]: I reverted it

File: domains/transfer/Transfer.API/API/Shared/Swagger/ConfigureSwaggerOptions.cs (hunk 13 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API/Shared/Swagger/ConfigureSwaggerOptions.cs:31]: It will enable the authorization button, in swagger ui in preview environments. @CodeReaper has approved the change

File: domains/transfer/Transfer.API/API/Cvr/Api/v2024_01_03/Controllers/CvrController.cs (hunk 9 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API/Cvr/Api/v2024_01_03/Controllers/CvrController.cs:25]: Changed

File: domains/transfer/Transfer.API/API/Cvr/Api/v2024_01_03/Controllers/CvrController.cs (hunk 11 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API/Cvr/Api/v2024_01_03/Controllers/CvrController.cs:35]: Nothing, the hit is ignored, as if it was never made

File: domains/transfer/Transfer.API/API/Transfer/Api/v2024_01_03/Dto/Responses/TransferAutomationStatus.cs (hunk 14 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API/Transfer/Api/v2024_01_03/Dto/Responses/TransferAutomationStatus.cs:3]: okay

File: domains/transfer/Transfer.API/API/Cvr/Api/v2024_01_03/Controllers/CvrController.cs (hunk 9 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API/Cvr/Api/v2024_01_03/Controllers/CvrController.cs]: https://github.com/Energinet-DataHub/energy-origin/pull/558/commits/893686fdbe2f715060dd6372f2b06652f4e0243f

File: domains/transfer/Transfer.API/API.UnitTests/Transfer/Api/Converters/Base64ConverterTests.cs (hunk 6 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API.UnitTests/Transfer/Api/Converters/Base64ConverterTests.cs:33]: Warning from xunit linter. will find alternative solution

File: domains/transfer/Transfer.API/API.IntegrationTests/API.IntegrationTests.csproj (hunk 10 above)
[the remark this answers is not in the document]
Comment (sahma19) [reply domains/transfer/Transfer.API/API.IntegrationTests/API.IntegrationTests.csproj:59]: Doesn't work. Keeping them there

File: domains/transfer/ClaimAutomation/Worker.IntegrationTests/Worker.IntegrationTests.csproj (hunk 4 above)
[the remark this answers is not in the document]
Comment (sahma19) [reply domains/transfer/ClaimAutomation/Worker.IntegrationTests/Worker.IntegrationTests.csproj]: Done:

image

File: domains/transfer/Transfer.API/API.UnitTests/Transfer/Api/Converters/Base64ConverterTests.cs (hunk 6 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API.UnitTests/Transfer/Api/Converters/Base64ConverterTests.cs:33]: I fixed it here:

https://github.com/Energinet-DataHub/energy-origin/pull/558/commits/aa903c41a8672c0e123784cf89cd7bb231b43fb1

[the remark this answers is not in the document]
Comment (sahma19) [reply domains/transfer/Transfer.API/API.UnitTests/Transfer/Api/Converters/Base64ConverterTests.cs:33]: fixed: https://github.com/Energinet-DataHub/energy-origin/pull/558/commits/aa903c41a8672c0e123784cf89cd7bb231b43fb1

File: domains/transfer/Transfer.API/API.IntegrationTests/API.IntegrationTests.csproj (hunk 10 above)
[the remark this answers is not in the document]
Comment (sahma19) [reply domains/transfer/Transfer.API/API.IntegrationTests/API.IntegrationTests.csproj:59]: I am resolving this

File: domains/transfer/Transfer.API/API/Cvr/Api/Clients/Cvr/CvrClient.cs (hunk 15 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API/Cvr/Api/Clients/Cvr/CvrClient.cs:20]: fixed here:

https://github.com/Energinet-DataHub/energy-origin/pull/558/commits/0ea9743cb222e5b66335ac4ed2e7677c66104832

File: domains/transfer/Transfer.API/API/Transfer/Api/v2024_01_03/Dto/Responses/TransferAutomationStatus.cs (hunk 14 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API/Transfer/Api/v2024_01_03/Dto/Responses/TransferAutomationStatus.cs]: fixed here:

https://github.com/Energinet-DataHub/energy-origin/pull/558/commits/0ea9743cb222e5b66335ac4ed2e7677c66104832

File: domains/transfer/Transfer.API/API/Cvr/Api/Clients/Cvr/CvrClient.cs (hunk 15 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API/Cvr/Api/Clients/Cvr/CvrClient.cs]: Done:

https://github.com/Energinet-DataHub/energy-origin/pull/558/commits/c171ad4ebc42b8efec6ee8a166d4362bc58e2068

File: domains/transfer/Transfer.API/API.IntegrationTests/Cvr/Api/v2024_01_03/Controllers/CvrMultipleResponseTests.cs [hunk 16]
```diff
@@ -0,0 +1,56 @@
...
+
+    public CvrMultipleResponseTests(TransferAgreementsApiWebApplicationFactory factory)
+    {
+        this.factory = factory;
+        server = WireMockServer.Start();
+        factory.CvrBaseUrl = server.Url!;
+    }
+
+    [Fact]
+    public async Task GetCvrCompany_WhenWrongPlusRight_ShouldReturnRightOnly()
```
[the remark this answers is not in the document]
Comment (sahma19) [reply domains/transfer/Transfer.API/API.IntegrationTests/Cvr/Api/v2024_01_03/Controllers/CvrMultipleResponseTests.cs]: Fixed:

https://github.com/Energinet-DataHub/energy-origin/pull/558/commits/4ccde03c72ab1cf1c4c647d82c3d40bd9b6381b2

[the remark this answers is not in the document]
Comment (sahma19) [reply domains/transfer/Transfer.API/API.IntegrationTests/Cvr/Api/v2024_01_03/Controllers/CvrMultipleResponseTests.cs]: Fixed: https://github.com/Energinet-DataHub/energy-origin/pull/558/commits/4ccde03c72ab1cf1c4c647d82c3d40bd9b6381b2

File: domains/transfer/Transfer.API/API/Cvr/Api/v2024_01_03/Controllers/CvrController.cs (hunk 2 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API/Cvr/Api/v2024_01_03/Controllers/CvrController.cs]: Done:

https://github.com/Energinet-DataHub/energy-origin/pull/558/commits/ade92237814c50ef674632ffd110dbbdbed85b75

File: domains/transfer/Transfer.API/API.IntegrationTests/Cvr/Api/v2024_01_03/Controllers/CvrMultipleResponseTests.cs (hunk 1 above)
Comment (ckr123) [reply domains/transfer/Transfer.API/API.IntegrationTests/Cvr/Api/v2024_01_03/Controllers/CvrMultipleResponseTests.cs:40]: Can you rename it to be self explaning? eg. cvr_multiple_companies.json

File: domains/transfer/Transfer.API/API.IntegrationTests/Cvr/Api/v2024_01_03/Controllers/CvrMultipleResponseTests.cs (hunk 8 above)
Comment (ckr123) [reply domains/transfer/Transfer.API/API.IntegrationTests/Cvr/Api/v2024_01_03/Controllers/CvrMultipleResponseTests.cs:17]: Please put this explanation as a comment

File: domains/transfer/Transfer.API/API/Cvr/Api/v2024_01_03/Controllers/CvrController.cs (hunk 11 above)
Comment (ckr123) [reply domains/transfer/Transfer.API/API/Cvr/Api/v2024_01_03/Controllers/CvrController.cs:40]: Will the api return 200 ok empty list?

File: domains/transfer/Transfer.API/API.IntegrationTests/SwaggerTests.GetSwaggerDocs_v20240103.verified.txt [hunk 17]
```diff
@@ -73,8 +73,8 @@
               }
             }
           },
-          "404": {
-            "description": "CVR company not found"
+          "400": {
```
Comment (ckr123) [inline domains/transfer/Transfer.API/API.IntegrationTests/SwaggerTests.GetSwaggerDocs_v20240103.verified.txt:76]: This should return a ValidationProblem

File: domains/transfer/Transfer.API/API.IntegrationTests/Cvr/Api/v2024_01_03/Controllers/CvrMultipleResponseTests.cs (hunk 1 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API.IntegrationTests/Cvr/Api/v2024_01_03/Controllers/CvrMultipleResponseTests.cs]: Yes:

https://github.com/Energinet-DataHub/energy-origin/pull/558/commits/5a4183c995c2ea621c69e2c1983c53e884cb8d17

File: domains/transfer/Transfer.API/API/Cvr/Api/v2024_01_03/Controllers/CvrController.cs [hunk 18]
```diff
@@ -0,0 +1,100 @@
...
+{
+    /// <summary>
+    /// Get CVR registered company information for multiple CVR numbers
+    /// </summary>
+    /// <response code="200">Successful operation</response>
+    /// <response code="400">A non-empty request body is required.</response>
+    [ProducesResponseType(typeof(CvrCompanyListResponse), 200)]
+    [ProducesResponseType(typeof(void), 400)]
+    [HttpPost]
+    public async Task<ActionResult<CvrCompanyListResponse>> GetCvrCompanies([FromBody] List<string> cvrNumbers)
```
Comment (ckr123) [inline domains/transfer/Transfer.API/API/Cvr/Api/v2024_01_03/Controllers/CvrController.cs:27]: the body should also be in a dto according to our guidelines

File: domains/transfer/Transfer.API/API/Cvr/Api/v2024_01_03/Controllers/CvrController.cs [hunk 19]
```diff
@@ -0,0 +1,100 @@
...
+[ApiController]
+[ApiVersion(20240103)]
+[Route("api/cvr")]
+public class CvrController(CvrClient client) : Controller
+{
+    /// <summary>
+    /// Get CVR registered company information for multiple CVR numbers
+    /// </summary>
+    /// <response code="200">Successful operation</response>
+    /// <response code="400">A non-empty request body is required.</response>
```
Comment (ckr123) [inline domains/transfer/Transfer.API/API/Cvr/Api/v2024_01_03/Controllers/CvrController.cs:23]: This should not be specific

File: domains/transfer/Transfer.API/API/Cvr/Api/v2024_01_03/Controllers/CvrController.cs (hunk 11 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API/Cvr/Api/v2024_01_03/Controllers/CvrController.cs:40]: Yes.

File: domains/transfer/Transfer.API/API/Cvr/Api/v2024_01_03/Controllers/CvrController.cs (hunk 19 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API/Cvr/Api/v2024_01_03/Controllers/CvrController.cs]: Changed:

https://github.com/Energinet-DataHub/energy-origin/pull/558/commits/f069c46be91fcec251339ce4157db8132dbc5c55

File: domains/transfer/Transfer.API/API/Cvr/Api/v2024_01_03/Controllers/CvrController.cs (hunk 18 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API/Cvr/Api/v2024_01_03/Controllers/CvrController.cs]: Created dto, and i am using it:

https://github.com/Energinet-DataHub/energy-origin/pull/558/commits/f069c46be91fcec251339ce4157db8132dbc5c55

File: domains/transfer/Transfer.API/API.IntegrationTests/SwaggerTests.GetSwaggerDocs_v20240103.verified.txt (hunk 17 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API.IntegrationTests/SwaggerTests.GetSwaggerDocs_v20240103.verified.txt:67]: It already does:

image

File: domains/transfer/Transfer.API/API.IntegrationTests/Cvr/Api/v2024_01_03/Controllers/CvrMultipleResponseTests.cs (hunk 8 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API.IntegrationTests/Cvr/Api/v2024_01_03/Controllers/CvrMultipleResponseTests.cs:18]: I've put a comment in the top of the PR

Comment (ckr123) [reply domains/transfer/Transfer.API/API.IntegrationTests/Cvr/Api/v2024_01_03/Controllers/CvrMultipleResponseTests.cs:18]: Sorry for not being clear. I ment to be more specific in the existing comment in the code on why it is not possible

Comment (sahma19) [reply domains/transfer/Transfer.API/API.IntegrationTests/Cvr/Api/v2024_01_03/Controllers/CvrMultipleResponseTests.cs]: Done:

https://github.com/Energinet-DataHub/energy-origin/pull/558/commits/e9a7d13a9eb6f54fb7dc358ac40b449cba881328

**Review by ckr123 — APPROVED** [review]
